### PR TITLE
ports/esp32: Skip validation of image on boot from deepsleep.

### PR DESCRIPTION
**Pull request by glenn20**
Add `CONFIG_BOOTLOADER_SKIP_VALIDATE_IN_DEEP_SLEEP=y` to `ports/esp32/boards/sdkconfig.base`. 

This cuts over 200ms from the time to boot from deepsleep on ESP32, ESP32-S2, ESP32-S3 and ESP32-C3 devices. This saving can lead to significant improvements in latency and energy efficiency for typical deepsleep-based applications, such as battery powered sensors.

This config flag is documented at: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/kconfig.html#config-bootloader-skip-validate-in-deep-sleep.

The image will continue to be validated on power-up boot or any other type of boot or reset. The validation is only skipped on wake from deepsleep.

Risks: The active OTA partition should not be changed before entering deepsleep with this option enabled or the new image will be booted without validation. 
- That is, after changing the OTA partition, the device should be hard reset to ensure the image is validated at least once before entering deepsleep. This seems a very low risk and the impact is low vs a significant improvement for battery powered deepsleep-based applications.
- In mitigation of this risk, this PR includes a note in the documentation for `esp32.Partition.set_boot()`.

Measurements and additional information on the latency and power savings is available at: https://github.com/glenn20/upy-esp32-experiments/tree/main/OptimisingMicropythonBootTime and https://github.com/orgs/micropython/discussions/10153.

### CODE REVIEW

**Comment by ThinkTransit** [comment]
I use this on all my builds and never had any issue, agree it is a low risk addition that should be considered for inclusion.

**Comment by stephanelsmith** [comment]
I would benefit from this PR as well, would love to see it included.

File: docs/library/esp32.rst [hunk 1]
```diff
@@ -114,6 +114,11 @@ methods to enable over-the-air (OTA) updates.
 
     Sets the partition as the boot partition.
 
+    **Note:** do not enter deepsleep (:func:`machine.deepsleep`) after changing
```
Comment (projectgus) [inline docs/library/esp32.rst:117]:
```suggestion
   .. note:: Do not enter deepsleep (:func:`machine.deepsleep`) after changing
```

Just a nitpick as it's possible to have Sphinx highlight the note. Thanks for documenting the extra limitation!

**Comment by projectgus** [comment]
This LGTM, useful optimisation for low power!

File: docs/library/esp32.rst (hunk 1 above)
Comment (glenn20) [reply docs/library/esp32.rst]: Fixed. Thanks - that's better.

**Comment by dpgeorge** [comment]
Thanks!

**Pull request merged**